modelstore/storage/aws.py

```diff
@@ -126,17 +131,26 @@ def _get_storage_location(self, meta: dict) -> str:
         return meta["prefix"]
 
     def _read_json_objects(self, path: str) -> list:
+        logger.debug("Listing files in: %s/%s", self.bucket_name, path)
         results = []
         objects = self.client.list_objects_v2(Bucket=self.bucket_name, Prefix=path)
         for version in objects.get("Contents", []):
-            if not version["Key"].endswith(".json"):
+            object_path = version["Key"]
+            if not object_path.endswith(".json"):
+                logger.debug("Skipping non-json file: %s", object_path)
+                continue
+            if os.path.split(object_path)[0] != path:
+                # We don't want to read files in a sub-prefix
+                logger.debug("Skipping file in sub-prefix: %s", object_path)
                 continue
```

Comment:
This was a bug. Model ids are stored in:

```
<bucket-name>/<root-prefix>/operatorai-model-store/<domain-name>/versions/*.json
```

And model ids that have been set to a specific state are also stored in:

```
<bucket-name>/<root-prefix>/operatorai-model-store/<domain-name>/versions/<state-name>/*.json
```

Without this line, listing model ids would read all of the models, including those that have been set to a specific state. This became problematic because I'm keeping 1 small JSON file to indicate that a model has been deleted.